keps/sig-node/3675-resource-plugin-manager/README.md

```diff
@@ -0,0 +1,920 @@
+[kubernetes/enhancements]: https://git.k8s.io/enhancements
+[kubernetes/kubernetes]: https://git.k8s.io/kubernetes
+[kubernetes/website]: https://git.k8s.io/website
+
+## Summary
+
+The authors have taken inspiration from the [CSI development](https://github.com/kubernetes/design-proposals-archive/blob/main/storage/container-storage-interface.md). 
+Kubernetes compute management is tightly integrated with the Kubelet and the 
+existing suite of resources managers including the topology manager, CPU manager, 
+memory manager, and device manager. While these managers have added functionality that 
+has addressed a varied set of use cases, they do present the community with several challenges.
+
+Adding a new capability to one of these managers is slow moving and difficult 
+due to their complex interactions, the level of prudence required given the potential
+impacts, the implied necessity to make the extension work in as many scenarios as
+possible, as well as the overhead heaped onto the sig-node maintainers. More 
+details on the challenges discussed with the community have been captured in 
+the [CPU Management Kubelet Use Cases & Current State document](https://docs.google.com/document/d/1U4jjRR7kw18Rllh-xpAaNTBcPsK5jl48ZAVo7KRqkJk).  
+
+As a result, adding optimizations necessary to support some more niche use cases, 
+to improve total cost of ownership for deployments with more exacting requirements 
+on resources or to provide a mechanism to progressively roll-out the benefits of 
+innovations related to newer CPU architectures is difficult and high cost to community resources.
+
+This proposal aims to address the challenges by introducing a new Container Compute 
+Interface (CCI). The CCI is conceptually equivalent to the CSI in that it will 
+delineate between the responsibility of the Kubelet and lower-level responsibility
+of compute focused plugins for resource assignment capabilities, such CPU and memory. 
+
+We are propsing an iterative implementation approach: in the first stage (alpha) we 
+propose to guarantee the exclusive operation of CCI Manager on cpu and memory resources
+through a feature gate and a requirement to have CPU Manager Policy as None. All pods will be handled 
+by the new CCI Manager. This will allow us to avoid state synchronization issues with cpu manager.
+To avoid the bootstrapping and driver-unavailability problems of CCI we propose to 
+use best-effort handling of those cases in the alpha state of the implementation augmented with additional logging. 
+
+In a later phase of the implementation we can provide a proper mechanism to handle driver unavailability cases. 
+Given the existing Kubelet CPU/memory manager capabilities and their usefulness 
+for many scenario, we would also like to consider if resource management for certain pods have to be handled without drivers and if we need 
+to associate specific pods with CCI Drivers (association mechanisms will be investigated in 
+later phase).
```

Comment:
Mixed use of plugins and drivers -- I think we should stick with one or the other. I have been referring to them as plugins in all of my comments until now, but if we decide to go with `driver`, we should very clearly spell out that that is what we are calling them, rather then just starting to use the term without explanation.

Reply:
Regardless of that, I think this text is out of place as it doesn't flow well from the previous paragraph into the next one. Please consider rewording / moving.

Reply:
done